### PR TITLE
[docs] updated bug report template

### PR DESCRIPTION
**Pull request by jameslamb**
I'd like to update the bug report template. I have seen the following things in recent issues that I think we should try to nudge people to avoid.

* ambiguous environment information
* using screenshots instead of code snippets

Some examples of "ambiguous environment information" from recent issues:

image

image

image

image

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/BUG_REPORT.md [hunk 1]
```diff
@@ -11,24 +11,64 @@ Please search your question on previous issues, Stack Overflow (https://stackove
...
 ## Environment info
 
+<!-- Fill out each of the below. If something is not applicable, put 'N/A' or leave it blank -->
+
 Operating System:
 
 CPU/GPU model:
 
-C++/Python/R version:
+C++ version:
```
Comment (StrikerRUS) [inline .github/ISSUE_TEMPLATE/BUG_REPORT.md:36]: I think compiler version will be more helpful for us.
```suggestion
C++ compiler version:
```

File: .github/ISSUE_TEMPLATE/BUG_REPORT.md [hunk 2]
```diff
@@ -11,24 +11,64 @@ Please search your question on previous issues, Stack Overflow (https://stackove
...
+-->
 
-<!-- Paste error log here -->
 
-## Reproducible examples
+## Reproducible example(s)
 
-<!-- Paste examples here -->
+<!--
+Paste examples log below.
```
Comment (StrikerRUS) [inline .github/ISSUE_TEMPLATE/BUG_REPORT.md:64]:
```suggestion
Paste examples code below.
```

File: .github/ISSUE_TEMPLATE/BUG_REPORT.md [hunk 3]
```diff
@@ -11,24 +11,64 @@ Please search your question on previous issues, Stack Overflow (https://stackove
...
+
+Please paste code and text logs instead of screenshots, so others
+can find this issue from search engines.
+
+If possible, try to provide a minimum working example that does not
+require access to proprietary data.
```
Comment (StrikerRUS) [inline .github/ISSUE_TEMPLATE/BUG_REPORT.md:70]:
```suggestion
require access to proprietary data: https://stackoverflow.com/help/minimal-reproducible-example.
```

**Comment by jameslamb** [comment]
> Thanks @jameslamb, great addition!
> But I'm afraid that the most people don't read these **long** comments. Also a lot of text with requirements scares some people and possibly discourages from posting an issue.

good point! What do you think about these simplifications? https://github.com/microsoft/LightGBM/pull/3032/commits/7299bde65bec2c8ef616b6747400a76ae89cac06

File: .github/ISSUE_TEMPLATE/BUG_REPORT.md [hunk 4]
```diff
@@ -7,28 +7,56 @@ about: Something isn't working as expected? Here is the right place to report.
 Please search your question on previous issues, Stack Overflow (https://stackoverflow.com/questions/tagged/lightgbm) or other search engines before you open a new one.
 -->
```
Comment (StrikerRUS) [inline .github/ISSUE_TEMPLATE/BUG_REPORT.md:8]:
```suggestion
Please search your question on previous issues, Stack Overflow (https://stackoverflow.com/questions/tagged/lightgbm) or other search engines before you open a new one.

Also, for bugs and unexpected issues, please check the latest master branch first.
-->
```

File: .github/ISSUE_TEMPLATE/BUG_REPORT.md [hunk 5]
```diff
@@ -7,28 +7,56 @@ about: Something isn't working as expected? Here is the right place to report.
...
+
+Other:
 
 <!-- Please check the latest master branch first -->
 LightGBM version or commit hash:
 
-## Error message
+## Error message and / or logs
+
+<!-- Paste error log below -->
```
Comment (StrikerRUS) [inline .github/ISSUE_TEMPLATE/BUG_REPORT.md:49]:
```suggestion
<!-- Paste error log below (please avoid screenshots, use raw text) -->
```

File: .github/ISSUE_TEMPLATE/BUG_REPORT.md [hunk 6]
```diff
@@ -7,28 +7,56 @@ about: Something isn't working as expected? Here is the right place to report.
...
 
-<!-- Paste error log here -->
 
-## Reproducible examples
+## Reproducible example(s)
 
-<!-- Paste examples here -->
+<!--
+Paste examples code below.
```
Comment (StrikerRUS) [inline .github/ISSUE_TEMPLATE/BUG_REPORT.md:55]:
```suggestion
Paste examples code below (please avoid screenshots, use raw text).
```

**Comment by jameslamb** [comment]
> @jameslamb WDYT about partly restoring some advices you proposed earlier but in shorter form?

the suggestions look good to me, thanks!

**Comment by jameslamb** [comment]
> @jameslamb Thanks a lot! LGTM!

thanks for the suggestions!

**Pull request merged**